File: ios/Amrosa/Amrosa/Data/DTOs/ProfileRecipeSummary.swift
import Foundation

/// A recipe shown on a chef's profile (F12). Read from the `shared_recipes` mirror.
struct ProfileRecipeSummary: Identifiable, Hashable {
    let recipeId: String
    let title: String
    let prepTimeMinutes: Int?
    let cookTimeMinutes: Int?
    let tags: [String]
    /// "friends" or "public" — drives the small tier badge on the card.
    let visibility: String

    var id: String { recipeId }
}

File: android/app/src/main/java/com/aerion/chefslist/domain/model/ProfileRecipeSummary.kt
package com.aerion.chefslist.domain.model

/**
 * Card-level data for a recipe shown on another chef's profile.
 * Sourced from the `shared_recipes` mirror (a co-chef sees the author's friends + public recipes).
 */
data class ProfileRecipeSummary(
    val recipeId: String,
    val title: String,
    val prepTimeMinutes: Int?,
    val cookTimeMinutes: Int?,
    val tags: List<String>,
    /** "friends" or "public" — drives the small tier badge on the card. */
    val visibility: String,
)
